### ISSUE TITLE
Fetchind Crash

### ISSUE THREAD

**Issue by d-voidheart** [victorrjimenezz/PS4-4PT#18]
Hey friend, first of all, thank you a lot for the 4PT. I'd like to advertise about some problems with 2.60 version. When I add long repositories (many lines of script in repo.yml file), it crashes 4PT many times. I don't know if the homebrew has item limit, but checking the logs I realised that your default repo load more than 300 lines of script, and the repo that I added has almost this value too. I've tried to remove your default repo but I can't. Other bug is about the delete files action, sometimes when I try to delete it crashes 4PT again.

**Comment by victorrjimenezz** [comment]
Hi!! I’m happy to see that you’re interested in the app. Regarding your concerns:

there should be no limit to the amount of repositories/packages loaded. Could you please share the logfile of a crashed session?

regarding the app crashing when deleting, does it crash when deleting a download, a repository, or both? Could you also share the logfile of a crashed session?

Thank you for sharing your concerns, this helps 4pt improve a lot!

**Comment by d-voidheart** [comment]
Hey friend. It crashes when I try to delete repositories. I realised something: the first time we open 4PT and it just has default repo (www.4pt-project.com) it works really well. Then we add a new repo and wait it loads, and it also works. But when we close 4PT and reopen later with the repos saved, it tries to load all repos at the same time and I GUESS that there is the error, I think it causes the crash when 4PT tries to load all repos at the same time and it gives some conflict. I don't have the log files anymore cause I was trying to solve in some ways. I realised when the 4PT already has more than 1 repos saved, and it crashes, we can solve this turning the internet off and opening the app. When it opens, it shows the registered repos links but without load, then if you delete the default repo (your 4PT repo) and then turn on the internet, the other repos loads correctly, and when we close 4PT and reopen this way, your repo appears in the last position and all repos above it loads well.

**Comment by victorrjimenezz** [comment]
> Hey friend. It crashes when I try to delete repositories. I realised something: the first time we open 4PT and it just has default repo ([www.4pt-project.com](http://www.4pt-project.com)) it works really well. Then we add a new repo and wait it loads, and it also works. But when we close 4PT and reopen later with the repos saved, it tries to load all repos at the same time and I GUESS that there is the error, I think it causes the crash when 4PT tries to load all repos at the same time and it gives some conflict. I don't have the log files anymore cause I was trying to solve in some ways. I realised when the 4PT already has more than 1 repos saved, and it crashes, we can solve this turning the internet off and opening the app. When it opens, it shows the registered repos links but without load, then if you delete the default repo (your 4PT repo) and then turn on the internet, the other repos loads correctly, and when we close 4PT and reopen this way, your repo appears in the last position and all repos above it loads well.

I understand, thank you for sharing!! I'll investigate this issues, could you please share the logfile located at /data/4PT/logs/ ?
Logs can provide useful information for fixing these bugs! :)

**Comment by d-voidheart** [comment]
[4PT Logs.zip](https://github.com/victorrjimenezz/PS4-4PT/files/8065065/4PT.Logs.zip) Here is friend.

**Comment by victorrjimenezz** [comment]
> [4PT Logs.zip](https://github.com/victorrjimenezz/PS4-4PT/files/8065065/4PT.Logs.zip) Here is friend.

Thank you! This provides very useful information. It seems to me that it crashes after failing to fetch a specific pkg. You can see in this line:
`[Mon Feb 14 19:58:13 2022] addPKG: FAILED TO FETCH: https://repo.pspkgs.workers.dev/1:/last2dlc.pkg`

**Comment by d-voidheart** [comment]
Nice! So, if the owner of this repo delete this PKG, 4PT will work correctly, right? The issue is on this PKG file, is that correct?

**Comment by victorrjimenezz** [comment]
> Nice! So, if the owner of this repo delete this PKG, 4PT will work correctly, right? The issue is on this PKG file, is that correct?

Correct. That _might_ be the issue. However, if this is it, the solution should not be to remove the pkg, the app should not crash even if the PKG is corrupted, it should just discard it! I'll work on this! :)
